### DISCUSSION TITLE
How to properly set fees with the new batch PublishStorageDeal feature?

### DISCUSSION THREAD

**Discussion by kernelogic** [filecoin-project/lotus#5541] [Storage Provider] [answered]
I am currently testing this new feature and after I imported 10 deals at once, zero deal appeared in the sectors list and when I examine the mpool there are 3 stuck messages. After I replace those messages and they turned out to be PublishStorageDeal error: PublishStorageDeals errored: WaitForPublishDeals exit code: SysErrOutOfGas(7).

This have never happened before to me. I have over 80 FIL in the balance and it still shows out of gas. Is there something wrong with the gas estimation for this new feature? Do I need to set MaxPublishDealsFee to something like 2 FIL?

My current fee settings on the miner is 
[Fees]
  MaxPreCommitGasFee = "0.3 FIL"
  MaxCommitGasFee = "0.3 FIL"
  MaxWindowPoStGasFee = "5 FIL"
  MaxPublishDealsFee = "0.5 FIL"
  MaxMarketBalanceAddFee = "0.5 FIL"

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
SysErrOutOfGas(7) is not related to how much FIL you have but to how much the gas-limit is

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
it will be tricky to the the max fee for batched messages since the gas-limit (and therefore the gas-feecap) may vary a lot if there are different amounts of deal proposals
 in the message

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment, marked as answer]
see https://filecoinproject.slack.com/archives/G01KU7G441Y/p1612662154060200
